java/src/main/java/com/adhoc/flight/QueryRunner.java

```diff
@@ -0,0 +1,254 @@
+import org.apache.arrow.flight.HeaderCallOption;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.util.AutoCloseables;
+import org.apache.arrow.util.Preconditions;
+
+import com.adhoc.flight.client.AdhocFlightClient;
+import com.adhoc.flight.utils.PrintUtils;
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParameterException;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Java Flight sample application that runs the specified query.
+ */
+public class QueryRunner {
+    private static final BufferAllocator BUFFER_ALLOCATOR = new RootAllocator(Integer.MAX_VALUE);
+    private static final CommandLineArguments ARGUMENTS = new CommandLineArguments();
+
+    private static final String CREATE_DEMO_TABLE =
+            "CREATE TABLE $scratch.dremio_flight_demo_table as select * from (VALUES(1,2,3),(4,5,6))";
+    private static final String DROP_DEMO_TABLE = "DROP TABLE $scratch.dremio_flight_demo_table";
+    private static final String SELECT_DEMO_TABLE = "SELECT * FROM dremio_flight_demo_table";
+    private static final String DEMO_TABLE_SCHEMA = "$scratch";
+
+    static class CommandLineArguments {
+        @Parameter(names = {"-host", "--hostname"},
+                description = "Dremio co-ordinator hostname")
+        public String host = "localhost";
+
+        @Parameter(names = {"-port", "--flightport"},
+                description = "Dremio flight server port")
+        public int port = 32010;
+
+        @Parameter(names = {"-user", "--username"},
+                description = "Dremio username")
+        public String user = "dremio";
+
+        @Parameter(names = {"-pass", "--password"},
+                description = "Dremio password")
+        public String pass = "dremio123";
+
+        @Parameter(names = {"-query", "--sqlQuery"},
+                description = "SQL query to test")
+        public String query = null;
+
+        @Parameter(names = {"-tls", "--tls"},
+                description = "Enable encrypted connection")
+        public boolean enableTls = false;
+
+        @Parameter(names = {"-kstpath", "--keyStorePath"},
+                description = "Path to the jks keystore")
+        public String keystorePath = null;
+
+        @Parameter(names = {"-kstpass", "--keyStorePassword"},
+                description = "The jks keystore password")
+        public String keystorePass = null;
+
+        @Parameter(names = {"-demo", "--runDemo"},
+                description = "A flag to to run a demo of querying the Dremio Flight Server Endpoint.")
+        public boolean runDemo = false;
+
+        @Parameter(names = {"-h", "--help"},
+                description = "show usage", help=true)
+        public boolean help = false;
+    }
+
+    /**
+     * Runs a self contained demo to authenticate and query a Dremio Flight Server Endpoint.
+     *
+     * @throws Exception
```

Comment:
We should have a description of why this is thrown.

Reply:
Addressed.